opm/autodiff/GeoProps.hpp

```diff
@@ -388,11 +391,26 @@ namespace Opm
                 const double* scaledFaceNormal = faceNormal;
 #endif
 
+                int cartesianCellIdx = AutoDiffGrid::globalCell(grid)[cellIdx];
+                auto cellCentroid = eclGrid->getCellCenter(cartesianCellIdx);
+
                 for (int indx = 0; indx < dim; ++indx) {
-                    const double Ci = Opm::UgGridHelpers::faceCentroid(grid, faceIdx)[indx] -
-                        Opm::UgGridHelpers::cellCentroidCoordinate(grid, cellIdx, indx);
+
+                    double Ci = Opm::UgGridHelpers::faceCentroid(grid, faceIdx)[indx];
+                    switch (indx) {
+                    case 0:
+                        Ci -= std::get<0>(cellCentroid);
+                        break;
+                    case 1:
+                        Ci -= std::get<1>(cellCentroid);
+                        break;
+                    case 2:
+                        Ci -= std::get<2>(cellCentroid);
+                        break;
+
+                    }
                     dist += Ci*Ci;
```

Comment:
I think we should change the return type to a std::array<double> in EclipseState. This should cause no pain downstream since std::array can be treated as a std::tuple (with homogenous types). I'll propose this there and we'll come back to this.